blocks/large-manual-promo-block/features/large-manual-promo/default.jsx

```diff
@@ -1,19 +1,136 @@
 import React from "react";
 import PropTypes from "@arc-fusion/prop-types";
-import { useFusionContext } from "fusion:context";
-import { LazyLoad, isServerSide } from "@wpmedia/engine-theme-sdk";
+import { useComponentContext, useFusionContext } from "fusion:context";
+import { useEditableContent } from "fusion:content";
+import { LazyLoad } from "@wpmedia/engine-theme-sdk";
+import {
+	formatURL,
+	isServerSide,
+	Overline,
+	MediaItem,
+	Grid,
+	Link,
+	Heading,
+	HeadingSection,
+	Image,
+	Paragraph,
+	Attribution,
+	Separator,
+	Date,
+	Stack,
+} from "@wpmedia/arc-themes-components";
 import { imageRatioCustomField } from "@wpmedia/resizer-image-block";
-import { LargePromoPresentation } from "@wpmedia/shared-styles";
+
+const BLOCK_CLASS_NAME = "b-large-manual-promo";
 
 const LargeManualPromo = ({ customFields }) => {
 	const { isAdmin } = useFusionContext();
-	if (customFields?.lazyLoad && isServerSide() && !isAdmin) {
+	const {
+		headline,
+		imagePosition,
+		imageURL,
+		lazyLoad,
+		caption,
+		credits,
+		subtitle,
+		linkURL,
+		newTab,
+		showHeadline,
+		showImage,
+		description,
+		overline,
+		showOverline,
+		showDescription,
+		overlineURL,
+		author,
+		dateTime,
+		dateString,
+		hideImageCaption,
+		hideImageCredits,
+		hideImageTitle,
+	} = customFields;
+	const shouldLazyLoad = lazyLoad && !isAdmin;
+	const { registerSuccessEvent } = useComponentContext();
+
+	if (shouldLazyLoad && isServerSide()) {
 		// On Server
 		return null;
 	}
+
+	const PromoImage = () => {
+		const { searchableField } = useEditableContent();
+
+		return showImage ? (
+			<MediaItem
+				caption={!hideImageCaption ? caption : null}
+				credit={!hideImageCredits ? credits : null}
+				title={!hideImageTitle ? subtitle : null}
+				{...searchableField("imageURL")}
+				suppressContentEditableWarning
+			>
+				<Image alt={headline} src={imageURL} searchableField />
+			</MediaItem>
+		) : null;
+	};
+
+	const PromoHeading = () =>
+		showHeadline ? (
+			<Heading className={`${BLOCK_CLASS_NAME}__heading`}>
```

Comment:
We do not need the classname applied to the `Heading` component as you can style the component via the `components: ()` map of the block